tests/unit/scm/test_scm.py

```diff
@@ -1,60 +1,61 @@
-from unittest import TestCase
+import pytest
 
-import mock
-
-from dvc.repo import Repo
 from dvc.repo.scm_context import scm_context
 from dvc.scm import NoSCM
 
 
-class TestScmContext(TestCase):
-    def setUp(self):
-        self.repo_mock = mock.Mock(spec=Repo)
-        self.scm_mock = mock.Mock(spec=NoSCM)
-        self.repo_mock.scm = self.scm_mock
+def test_scm_context(dvc, mocker):
+    scm = dvc.scm
+    m = mocker.patch.object(
+        scm, "track_file_changes", wraps=scm.track_file_changes
```

Comment:
Cannot use `spy` because of the `__getattr__` hack:
https://github.com/iterative/dvc/blob/7b591083792bd73236fd684cf513a625c8bbee95/dvc/scm/__init__.py#L9-L11